packages/grub2/config/grub.cfg

```diff
@@ -1,35 +1,18 @@
 set timeout=10
 
-# Load custom env file
 set env_file="/grubenv"
-search --no-floppy --file --set=env_blk "${env_file}"
-
-# Load custom env file
 set oem_env_file="/grub_oem_env"
-search --no-floppy --file --set=oem_blk "${oem_env_file}"
-
-# Load custom config file
-set custom_menu="/grubmenu"
-search --no-floppy --file --set=menu_blk "${custom_menu}"
-
-# Load custom config file
-set custom="/grubcustom"
-search --no-floppy --file --set=custom_blk "${custom}"
-
-if [ "${oem_blk}" ] ; then
-  load_env -f "(${oem_blk})${oem_env_file}"
-fi
+set custom_file="/grubcustom"
 
-if [ "${env_blk}" ] ; then
-  load_env -f "(${env_blk})${env_file}"
-fi
+load_env -f "${oem_env_file}"
+load_env -f "${env_file}"
```

Comment:
on a closer look, I believe the last variables file to load should be `/grub_oem_env`, so we ensure no custom variables from grubenv are overwriting those. `grubenv` should be the RW file stored in `/oem`. Also we should check the file exists before reading it.